docs/source/tertiary/whole-genome-regression.rst

```diff
@@ -0,0 +1,352 @@
+  also include a column ``values`` containing a numeric representation of each genotype.
+- ``sample_ids``: List of sample IDs. Can be created by applying ``glow.wgr.functions.get_sample_ids`` to a genotype
+  DataFrame.
+- ``variants_per_block``: Number of variants to include per block.
+- ``sample_block_count``: Number of sample blocks to create.
+
+Return
+======
+
+The function returns a block genotype matrix and a sample block mapping.
+
+Block genotype matrix
+---------------------
+
+If we imagine the block genotype matrix conceptually, we think of an *NxM* matrix *X* where each row *n* represents an
+individual sample, each column *m* represents a variant, and each cell *(n, m)* contains a genotype value for sample *n*
+at variant *m*.  We then imagine laying a coarse grid on top of this matrix such that matrix cells within the same
+coarse grid cell are all assigned to the same block *x*.  Each block *x* is indexed by a sample block ID (corresponding
+to a list of rows belonging to the block) and a header block ID (corresponding to a list of columns belonging to the
+block).  The sample block IDs are generally just integers 0 through the number of sample blocks.  The header block IDs
+are strings of the form 'chr_C_block_B', which refers to the Bth block on chromosome C.  The Spark DataFrame
+representing this block matrix can be thought of as the transpose of each block *xT* all stacked one atop another.  Each
+row represents the values from a particular column from *X*, for the samples corresponding to a particular sample block.
+The fields in the DataFrame are:
+
+- ``header``: A column name in the conceptual matrix *X*.
+- ``size``: The number of individuals in the sample block for the row.
+- ``values``: Genotype values for this header in this sample block.  If the matrix is sparse, contains only non-zero values.
+- ``header_block``: An ID assigned to the block *x* containing this header.
+- ``sample_block``: An ID assigned to the block *x* containing the group of samples represented on this row.
+- ``position``:  An integer assigned to this header that specifies the correct sort order for the headers in this block.
+- ``mu``: The mean of the genotype calls for this header.
+- ``sig``: The standard deviation of the genotype calls for this header.
+
+Sample block mapping
+--------------------
+
+The sample block mapping consists of key-value pairs, where each key is a sample block ID and each value is a list of
+sample IDs contained in that sample block. The order of these IDs match the order of the ``values`` arrays in the block
+genotype DataFrame.
+
+Example
+=======
+
+.. code-block:: python
+
+    from glow.wgr.linear_model import RidgeReducer, RidgeRegression
+    from glow.wgr.functions import block_variants_and_samples, get_sample_ids
+    import numpy as np
+    from pyspark.sql.functions import col, lit
+
+    variants_per_block = 5
```

Comment:
what do we recommend for variants_per_block and sample_block_count and why?

Reply:
I changed the params to include a recommendation; we have had good results with 1k variants per block and 10 sample blocks over a wide range of sample counts.